modules/twinklexfd.js

```diff
@@ -346,53 +346,43 @@ Twinkle.xfd.callback.change_category = function twinklexfdCallbackChangeCategory
-				'Geographic/Americas/USA': ['United States of America', 'Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia (U.S. state)', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington', 'Washington, D.C.', 'West Virginia', 'Wisconsin', 'Wyoming'],
-				'Geographic/Unsorted': ['Islands']
-			};
-
-			var delsort = work_area.append({
+			work_area.append({
 				type: 'select',
 				multiple: true,
 				name: 'delsortCats',
 				label: 'Choose deletion sorting categories:',
 				tooltip: 'Select a few categories that are specifically relevant to the subject of the article. Be as precise as possible; categories like People and USA should only be used when no other categories apply.'
 			});
 
-			$.each(delsortCategories, function(groupname, list) {
-				var group = delsort.append({ type: 'optgroup', label: groupname });
-				list.forEach(function(item) {
-					group.append({ type: 'option', label: item, value: item });
-				});
-			});
+			// delsort categories taken from [[Wikipedia:WikiProject Deletion sorting/Computer-readable.json]]
+			var query = {
+				action: 'parse',
+				format: 'json',
+				page: 'Wikipedia:WikiProject_Deletion_sorting/Computer-readable.json',
+				prop: 'wikitext',
+				smaxage: '86400', // cache for 1 day
+				maxage: '86400' // cache for 1 day
+			};
+			new Morebits.wiki.api('Get JSON list of deletion sorting categories', query, function(apiobj) {
+				try {
+					var wikitext = apiobj.getResponse().parse.wikitext;
+					var delsortCategories = JSON.parse(wikitext);
+					var select = $('[name="delsortCats"]');
+					var escapeDoubleQuotes = function(s) {
+						return s.replace(/"/g, '&quot;');
+					};
+					$.each(delsortCategories, function(groupname, list) {
+						groupname = escapeDoubleQuotes(groupname);
+						var delsortCat = $(select).append('<optgroup label="' + groupname + '">');
+						list.forEach(function(item) {
+							item = escapeDoubleQuotes(item);
+							$(delsortCat).append('<option value="' + item + '" label="' + item + '">' + item + '</option>');
+						});
+					});
+				} catch (err) {
+					// If unable to pull usable Deletion Sort categories, just give up and exit this function. The user won't be able to pick any categories, but that won't stop the AFD nomination.
```

Comment:
The behaviour you describe in the comment is ideal, but it doesn't actually require the try–catch to be present. 

It can be removed since it only prevents the warning with the error trace from being logged to the console, which would have been be helpful for debugging.